### PR TITLE
PBFT threading

### PR DESCRIPTION
**Pull request by ashamis**
Part of #774 
Still some testing to go but fairly close to the PR

- This run the PBFT executions threads on thread 1 if that is available

### CODE REVIEW

**Comment by cimetrics** [comment]
threading_8@[5451 aka 20200228.29](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=5451&view=results) vs master ewma over 14 builds from [5233](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=5233&view=results) to [5437](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=5437&view=results)
images

File: src/enclave/rpchandler.h [hunk 1]
```diff
@@ -32,16 +32,15 @@ namespace enclave
...
     virtual ProcessPbftResp process_pbft(
-      std::shared_ptr<enclave::RpcContext> ctx, bool include_merkle_roots) = 0;
+      std::shared_ptr<enclave::RpcContext> ctx) = 0;
     virtual ProcessPbftResp process_pbft(
       std::shared_ptr<enclave::RpcContext>,
       ccf::Store::Tx& tx,
-      bool playback,
-      bool include_merkle_roots) = 0;
+      bool playback) = 0;
+    virtual crypto::Sha256Hash GetMerkleRoot() = 0;
```
Comment (ashamis) [inline src/enclave/rpchandler.h]: naming

File: src/consensus/pbft/libbyz/types.h [hunk 2]
```diff
@@ -48,7 +48,11 @@ static const uint32_t MERKLE_ROOT_SIZE = 32;
 struct ByzInfo
 {
   std::array<uint8_t, MERKLE_ROOT_SIZE> replicated_state_merkle_root;
-  int64_t ctx;
+  int64_t ctx = INT64_MIN;
+  void (*cb)(void* ctx) = nullptr;
```
Comment (olgavrou) [inline src/consensus/pbft/libbyz/types.h]: Does the callback ctx have to be a void pointer here?

Comment (ashamis) [reply src/consensus/pbft/libbyz/types.h]: no the ctx is not a pointer that is that actual context

**Pull request merged**